### PR TITLE
Fix groupWith docs

### LINKED ISSUE #1728: Error in R.groupWith() documentation
http://ramdajs.com/0.21.0/docs/#groupWith

- Wrong return value in the first example;
- Missed "R." prefix in the first and second examples;
- Broken indentation;
- Missed lovely "=>" prefix before returned value in the comments;

So the correct code snippet should be:
```
R.groupWith(R.equals, [0, 1, 1, 2, 3, 5, 8, 13, 21])
//=> [[0], [1, 1], [2], [3], [5], [8], [13], [21]]

R.groupWith((a, b) => a % 2 === b % 2, [0, 1, 1, 2, 3, 5, 8, 13, 21])
//=> [[0], [1, 1], [2], [3, 5], [8], [13, 21]]

R.groupWith(R.eqBy(isVowel), 'aestiou')
//=> ['ae', 'st', 'iou']
```

### PR DESCRIPTION
**Pull request by svozza**
Fixes #1728.

### CODE REVIEW

**Comment by buzzdecafe** [comment]
wow that was fast -- thanks @svozza

**Comment by svozza** [comment]
Ha. I was actually working with this function this morning so I figured I may as well.

File: src/groupWith.js [hunk 1]
```diff
@@ -8,23 +8,23 @@ var _curry2 = require('./internal/_curry2');
  * @memberOf R
  * @since v0.21.0
  * @category List
- * @sig (a, a -> Boolean) -> [a] -> [[a]]
+ * @sig ((a, a) → Boolean) → [a] → [[a]]
```
Comment (davidchambers) [inline src/groupWith.js]: Let's stick to `->` rather than `→`, in the source code at least.

Comment (CrossEye) [inline src/groupWith.js]: I'd actually rather keep them in the output as well, although I wouldn't mind if the font offered a ligature for the characters.

Comment (benperez) [inline src/groupWith.js]: Somebody showed me this yesterday, not sure if this is what you're looking for:

https://github.com/i-tu/Hasklig

Comment (CrossEye) [inline src/groupWith.js]: That looks like exactly what I want.  Anyone here have experience building fonts?

Comment (benperez) [inline src/groupWith.js]: @CrisFeo showed me this font package and mentioned that he might have some interest in contributing.

**Pull request merged**